libsplinter/src/node_registry/mod.rs

```diff
@@ -28,24 +28,65 @@ pub struct Node {
     pub metadata: HashMap<String, String>,
 }
 
+/// A predicate on a key/value pair in a Node's metadata table.
+///
+/// Each variant is an operator, and supplies a tuple representing a key/ value pair. It is applied
+/// by the comparison operator on the valued found at the given key (the first item in the tuple)
+/// against the predicate's value  (the second item in the tuple).
+///
+/// If the item is missing in a node's metadata table, the predicate returns false.
+pub enum MetadataPredicate {
+    /// Applies the `==` operator for
+    Eq(String, String),
+    Ne(String, String),
+    Gt(String, String),
+    Ge(String, String),
+    Lt(String, String),
+    Le(String, String),
+}
+
+impl MetadataPredicate {
+    /// Apply this predicate against a given node.
+    pub fn matches(&self, node: &Node) -> bool {
```

Comment:
Would more sense semantically for this method to be part of the `Node` struct, so you'd have `node.matches(predicate)` rather than `predicate.matches(node)`.

Reply:
Possibly, but this method is more of a convenience for in-memory usage.  Other implementations will use the enum to generate SQL statements, for example.

Reply:
Perhaps the name of the function could be changed from `matches` to `apply`